### PR TITLE
DX-2224: Added StarkEx Transfer functionality

### PR DESCRIPTION
**Pull request by suluvai**
Added following StarkEx functionalities
- Transfer
- BatchNftTtansfer

Tiny bit of clean up.

### CODE REVIEW

File: Source/Immutable/Private/Immutable/Actions/ImtblPassportImxTransferAsyncAction.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	BlueprintNode->TokenAddress = TokenAddress;
+	return BlueprintNode;
+}
+
+
+void UImmutablePassportImxTransferAsyncAction::Activate()
+{
+	if (!WorldContextObject || !WorldContextObject->GetWorld())
+	{
+		const FString Err = "GetEmail failed due to missing world or world context object.";
```
Comment (dom-murray) [inline Source/Immutable/Private/Immutable/Actions/ImtblPassportImxTransferAsyncAction.cpp:30]: Can we change GetEmail to ImxTransfer.

File: Source/Immutable/Private/Immutable/Actions/ImtblPassportImxBatchNftTransferAsyncAction.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	BlueprintNode->NftTransferDetails = NftTransferDetails;
+	return BlueprintNode;
+}
+
+
+void UImmutablePassportImxBatchNftTransferAsyncAction::Activate()
+{
+	if (!WorldContextObject || !WorldContextObject->GetWorld())
+	{
+		const FString Err = "GetEmail failed due to missing world or world context object.";
```
Comment (dom-murray) [inline Source/Immutable/Private/Immutable/Actions/ImtblPassportImxBatchNftTransferAsyncAction.cpp:25]: Can we change GetEmail to ImxBatchNftTransfer

File: Source/Immutable/Private/Immutable/Actions/ImtblPassportImxBatchNftTransferAsyncAction.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		IMTBL_WARN("Error: %s", *Err)
+		TArray<FString> TransferIds;
+		Failed.Broadcast(Err, TransferIds);
+		return;
+	}
+
+	GetSubsystem()->WhenReady(this, &UImmutablePassportImxBatchNftTransferAsyncAction::DoTransfer);//, /* timoutSec */ 15.0f);
```
Comment (dom-murray) [inline Source/Immutable/Private/Immutable/Actions/ImtblPassportImxBatchNftTransferAsyncAction.cpp:32]: Can you remove the commented out code here.

File: Source/Immutable/Private/Immutable/Actions/ImtblPassportImxTransferAsyncAction.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		IMTBL_WARN("Error: %s", *Err)
+		Failed.Broadcast(Err, TEXT(""));
+		return;
+	}
+
+	GetSubsystem()->WhenReady(this, &UImmutablePassportImxTransferAsyncAction::DoTransfer);//, /* timoutSec */ 15.0f);
```
Comment (dom-murray) [inline Source/Immutable/Private/Immutable/Actions/ImtblPassportImxTransferAsyncAction.cpp:36]: Can you remove the commented out code here.

File: Source/Immutable/Private/Immutable/ImmutablePassport.cpp [hunk 5]
```diff
@@ -610,7 +590,57 @@ void UImmutablePassport::OnGetEmailResponse(FImtblJSResponse Response)
...
-        ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{bSuccess, Msg});
+        ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{bSuccess, Msg, Response});
+    }
+}
+
+void UImmutablePassport::OnTransferResponse(FImtblJSResponse Response)
+{
+    if (auto ResponseDelegate = GetResponseDelegate(Response))
+    {
+        auto TrasferResponse = JsonObjectToUStruct<FImxTransferResponse>(Response.JsonObject);
```
Comment (dom-murray) [inline Source/Immutable/Private/Immutable/ImmutablePassport.cpp:601]: Typo in variable name.

**Review by dom-murray — CHANGES_REQUESTED** [review]

**Review by dom-murray — APPROVED** [review]

**Review by nattb8 — APPROVED** [review]

**Pull request merged**